pkg/controller/database/mysqlserver/managed.go

```diff
@@ -103,102 +105,98 @@ func (c *connecter) Connect(ctx context.Context, mg resource.Managed) (resource.
 	if err := c.client.Get(ctx, n, s); err != nil {
 		return nil, errors.Wrap(err, errGetProviderSecret)
 	}
-	client, err := c.newClientFn(s.Data[p.Spec.Secret.Key])
-	return &external{client: client, newPasswordFn: util.GeneratePassword}, errors.Wrap(err, errNewClient)
+	sqlClient, err := c.newClientFn(s.Data[p.Spec.Secret.Key])
+	return &external{kube: c.client, client: sqlClient, newPasswordFn: util.GeneratePassword}, errors.Wrap(err, errNewClient)
 }
 
 type external struct {
-	client        azure.MySQLServerAPI
+	kube          client.Client
+	client        database.MySQLServerAPI
 	newPasswordFn func(len int) (password string, err error)
 }
 
 func (e *external) Observe(ctx context.Context, mg resource.Managed) (resource.ExternalObservation, error) {
-	s, ok := mg.(*v1alpha3.MySQLServer)
+	cr, ok := mg.(*v1alpha3.MySQLServer)
```

Comment:
I notice you use `cr` as the variable name for most managed resources. What does `cr` stand for?

Reply:
It stands for `Custom Resource`